src/UI/Monitor.cs

```diff
@@ -312,16 +304,55 @@ void Render_menu(Vessel v)
 			GUILayout.Space(Styles.ScaleFloat(10.0f));
 		}
 
-
 		void Render_filter()
 		{
 			// show the group filter
 			GUILayout.BeginHorizontal(Styles.entry_container);
+
+			Render_TypeFilterButon(VesselType.Probe);
+			Render_TypeFilterButon(VesselType.Rover);
+			Render_TypeFilterButon(VesselType.Lander);
+			Render_TypeFilterButon(VesselType.Ship);
+			Render_TypeFilterButon(VesselType.Station);
+			Render_TypeFilterButon(VesselType.Base);
+			Render_TypeFilterButon(VesselType.Plane);
+			Render_TypeFilterButon(VesselType.Relay);
+			Render_TypeFilterButon(VesselType.EVA);
+
 			filter = Lib.TextFieldPlaceholder("Kerbalism_filter", filter, filter_placeholder, filter_style).ToUpper();
 			GUILayout.EndHorizontal();
 			GUILayout.Space(Styles.ScaleFloat(10.0f));
 		}
 
+		void Render_TypeFilterButon(VesselType type) {
+			Boolean isFiltered = filter_types.Contains(type);
+			GUILayout.Label(new GUIContent(" ", getVesselTypeIcon(type, isFiltered), type.displayDescription()), config_style);
+			if (Lib.IsClicked()) {
+				if(isFiltered) filter_types.Remove(type);
+				else filter_types.Add(type);
+			}
+		}
+
+		Texture2D getVesselTypeIcon(VesselType type, Boolean disabled = false)
+		{
+			// TODO use proper ship type icons here. There should be a way to fetch those from
+			// KSP, but I don't know how or where to get them
+			switch(type) {
+				// this is what it will look like once someone has found the correct icons...
+				// case VesselType.Base: return disabled ? Icons.base_grey : Icons.base_white;
+
+				case VesselType.Base: return disabled ? Icons.battery_yellow : Icons.battery_white;
```

Comment:
I need someone to actually provide the correct icons here, these are just placeholders. However, as I currently a) don't see any icons at all in my own builds (mac...) and b) wouldn't know where to get those icons from (I'm sure they're somewhere readily available within the depths of the KSP/Unity API...) I can't do that myself.

Help, anyone?

Reply:
This would fix #186

Reply:
I'll have a go later tonight if I have some spare time but will more likely be Saturday